h/security/policy/top_level.py

```diff
@@ -24,10 +24,16 @@ def forget(self, request, **kw):
     def identity(self, request) -> Identity | None:
         return self._identity_cache.get_or_create(request)
 
+    def authenticated_userid(self, request):
+        return get_subpolicy(request).authenticated_userid(request)
+
     def remember(self, request, userid, **kw):
         self._identity_cache.clear(request)
         return get_subpolicy(request).remember(request, userid, **kw)
 
+    def permits(self, request, context, permission) -> Allowed | Denied:
+        return get_subpolicy(request).permits(request, context, permission)
```

Comment:
Unlike with the various sub-policies, `TopLevelPolicy`'s `authenticated_userid()` and `permits()` methods *do not* delegate to the new helpers. Instead, they delegate to the applicable sub-policy the same way that `TopLevelPolicy`'s other methods do. I think this makes sense: `TopLevelPolicy`'s only job is to choose the applicable sub-policy and delegate all calls to the sub-policy. It's the sub-policy's business how those methods are implemented (and whether or not they work by calling the new helpers). This also makes the behaviour of all of `TopLevelPolicy`'s methods consistent with each other.